grr/server/grr_response_server/gui/ui/lib/api_translation/primitive.ts

```diff
@@ -71,3 +71,72 @@ export function createUnknownObject(anyObject?: AnyObject): unknown|undefined {
   delete result['@type'];
   return result;
 }
+
+/**
+ * Decodes a encoded base64 string into a byte array.
+ * On error, returns an empty array.
+ */
+export function decodeBase64(encodedString?: string): Uint8Array {
+  if (encodedString === undefined) {
+    return new Uint8Array();
+  }
+
+  let decodedString;
+  try {
+    decodedString = atob(encodedString);
+  } catch (err) {
+    return new Uint8Array();
+  }
+  const byteArray = new Uint8Array(decodedString.length);
+  for (let i = 0; i < decodedString.length; i++) {
+    byteArray[i] = decodedString.charCodeAt(i);
+  }
+
+  return byteArray;
+}
+
+/** Returns the uppercase hex representation of the first byte of the provided number */
```

Comment:
Here's a trap: `number & 0xFF` zeroes out everything except the least-significant byte. The functions documentation says the "first byte" is used and the function name does not specify which byte at all.

One solution could be to write a function that shows the hexadecimal version of the whole input and remove the padding logic from the function. This could simplify the code, because three logic pieces (truncating number, hex-conversion, and padding) are split.

Reply:
I'll update the function name and the documentation to better represent the implementation